examples/pytorch/elastic/imagenet/Dockerfile

```diff
@@ -1,10 +1,10 @@
-ARG BASE_IMAGE=pytorch/pytorch:1.13.1-cuda11.6-cudnn8-runtime
+ARG BASE_IMAGE=pytorch/pytorch:2.2.1-cuda12.1-cudnn8-runtime
 FROM $BASE_IMAGE
 
 WORKDIR /workspace
 
 # download imagenet tiny for data
-RUN apt-get -q update && apt-get -q install -y wget unzip
+RUN apt-get -q update && apt-get -q install -y wget unzip g++
```

Comment:
If you don't install `g++`, any error happen?

Reply:
Yes. I got following error messages when I ran container built without g++, both CPU and GPU backend.
```
# On CPU
nvalidCxxCompiler: No working C++ compiler found in torch._inductor.config.cpp.cxx: (None, 'g++')

# On GPU
Failed to find C compiler. Please specify via CC environment variable
```

And after adding the line, I solved this issue.
It seems to be due to the JIT-compile in `torch.compile`.